From faa4fce8367f6f12b641ef47ef4427f61a711121 Mon Sep 17 00:00:00 2001
From: Masatake Iwasaki <iwasakims@apache.org>
Date: Sat, 13 Feb 2016 14:59:38 +0900
Subject: [PATCH 1710/2748] HADOOP-12589. Fix intermittent test failure of
 TestCopyPreserveFlag (iwasakims)

(cherry picked from commit 36ba5cac37dee6b60e607401ac06d263e4ac5d8f)
(cherry picked from commit 39970794d76e267adc8297d2604a4135b0c82e25)

Change-Id: Id9b86560e6a4a06833e2ad48f26edf82c1f5622b
---
 .../fs/contract/localfs/LocalFSContract.java       |    6 +++---
 .../hadoop/fs/shell/TestCopyPreserveFlag.java      |    5 ++---
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/contract/localfs/LocalFSContract.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/contract/localfs/LocalFSContract.java
index 8053335..74906fa 100644
--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/contract/localfs/LocalFSContract.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/contract/localfs/LocalFSContract.java
@@ -20,6 +20,7 @@
 
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileSystemTestHelper;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.fs.contract.AbstractFSContract;
 import org.apache.hadoop.fs.contract.ContractOptions;
@@ -39,9 +40,8 @@
 public class LocalFSContract extends AbstractFSContract {
 
   public static final String CONTRACT_XML = "contract/localfs.xml";
-  public static final String SYSPROP_TEST_BUILD_DATA = "test.build.data";
-  public static final String DEFAULT_TEST_BUILD_DATA_DIR = "test/build/data";
   private FileSystem fs;
+  private String testDataDir = new FileSystemTestHelper().getTestRootDir();
 
   public LocalFSContract(Configuration conf) {
     super(conf);
@@ -111,6 +111,6 @@ public Path getTestPath() {
    * @return the directory for test data
    */
   protected String getTestDataDir() {
-    return System.getProperty(SYSPROP_TEST_BUILD_DATA, DEFAULT_TEST_BUILD_DATA_DIR);
+    return testDataDir;
   }
 }
diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/shell/TestCopyPreserveFlag.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/shell/TestCopyPreserveFlag.java
index ecfb5a5..f9827ca 100644
--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/shell/TestCopyPreserveFlag.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/shell/TestCopyPreserveFlag.java
@@ -25,6 +25,7 @@
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FSDataOutputStream;
 import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileSystemTestHelper;
 import org.apache.hadoop.fs.LocalFileSystem;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.fs.permission.FsAction;
@@ -54,9 +55,7 @@ public void initialize() throws Exception {
     conf = new Configuration(false);
     conf.set("fs.file.impl", LocalFileSystem.class.getName());
     fs = FileSystem.getLocal(conf);
-    testDir = new Path(
-        System.getProperty("test.build.data", "build/test/data") + "/testStat"
-    );
+    testDir = new FileSystemTestHelper().getTestRootPath(fs);
     // don't want scheme on the path, just an absolute path
     testDir = new Path(fs.makeQualified(testDir).toUri().getPath());
 
-- 
1.7.9.5

